monai/bundle/workflows.py

```diff
@@ -226,6 +242,138 @@ def check_properties(self) -> list[str] | None:
+
+    @deprecated_arg(
+        "workflow",
+        since="1.2",
+        removed="1.5",
+        new_name="workflow_type",
+        msg_suffix="please use `workflow_type` instead.",
+    )
+    def __init__(
+        self,
+        workflow_type: str | None = None,
+        workflow: str | None = None,
+        properties_path: PathLike | None = None,
+        config_file: str | Sequence[str] | None = None,
+        meta_file: str | Sequence[str] | None = None,
+        logging_file: str | None = None,
+        **override: Any,
+    ):
+        meta_file = str(Path(os.getcwd()) / "metadata.json") if meta_file is None else meta_file
+        super().__init__(
+            workflow_type=workflow_type,
+            workflow=workflow,
+            properties_path=properties_path,
+            meta_file=meta_file,
+            logging_file=logging_file,
+        )
+        self._props: dict = {}
+        self._set_props: dict = {}
+        self.parser = ConfigParser()
+        if config_file is not None:
+            self.parser.read_config(f=config_file)
+        if self.meta_file is not None:
+            self.parser.read_meta(f=self.meta_file)
+
+        # the rest key-values in the _args are to override config content
+        self.parser.update(pairs=override)
+
+    def initialize(self, *args: Any, **kwargs: Any) -> Any:
+        """
+        Initialize the bundle workflow before running.
+        """
+        self._props = {}
+
+    @abstractmethod
+    def run(self, *args: Any, **kwargs: Any) -> Any:
+        """
+        Run the bundle workflow, it can be a training, evaluation or inference.
+
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def finalize(self, *args: Any, **kwargs: Any) -> Any:
+        """
+        Finalize step after the running of bundle workflow.
+
+        """
+        raise NotImplementedError()
+
+    def _get_property(self, name: str, property: dict) -> Any:
+        """
+        With specified property name and information, get the expected property value.
+        If the property is already generated, return from the bucket directly.
+        If user explicitly set the property, return it directly.
+        Otherwise, generate the expected property as a class private property with prefix "_".
+
+        Args:
+            name: the name of target property.
+            property: other information for the target property, defined in `TrainProperties` or `InferProperties`.
+        """
+        value = None
+        id = self.properties.get(name, None).get(BundlePropertyConfig.ID, None)
+        if name in self._set_props:
+            value = self._set_props[name]
+            self._props[name] = value
+        elif name in self._props:
+            value = self._props[name]
+        elif name in self.parser.config[self.parser.meta_key]:  # type: ignore[index]
+            value = self.parser.__getitem__(id)
```

Comment:
```suggestion
            value = self.parser[id]
```
This is equivalent here?

Reply:
Yes, thanks!